### PR TITLE
#416 Check licenses in PR Builder

### PR DESCRIPTION
**Pull request by symphony-thibault**
### Issue
https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/issues/416

### Description
A [Gradle plugin](https://github.com/jk1/Gradle-License-Report) has been added in order to validate licenses during PR Builder execution. 

By default, all kind of licenses are excluded by the plugin. We had to define allowed licences in file `.circleci/allowed-licenses.json`: 
```json
{
    "allowedLicenses": [
        {
            "moduleLicense": ".*Apache.*2.0",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "Bouncy Castle Licence",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "EPL 2.0",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "CC0",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "Eclipse Public License.*",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "MIT License",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "GNU.*General Public License.*",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "GPL2 w/ CPE",
            "moduleName": ".*"
        },
        {
            "moduleLicense": "Prior BSD License",
            "moduleName": ".*"
        },
        {
            "moduleName": "org.springframework.boot:spring-boot-dependencies"
        }
    ]
}

```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -55,6 +55,13 @@ jobs:
             find . -type d -regex ".*/build/reports/jacoco/test/html" -exec cp --parents -r {} ~/code-coverage/jacoco/ \;
           when: always
 
+      - run:
+          name: Save license check results
+          command: |
+            mkdir -p ~/reports/depedency-license/
```
Comment (symphony-youri) [inline .circleci/config.yml:61]: typo in the name of the folder?

File: build.gradle [hunk 2]
```diff
@@ -54,3 +56,7 @@ snyk {
     arguments = '--all-sub-projects'
     severity = 'high'
 }
+
+licenseReport {
+    allowedLicensesFile = new File("$projectDir/.circleci/allowed-licenses.json")
```
Comment (symphony-youri) [inline build.gradle:61]: as it is more specific to Gradle I would perhaps move it to buildSrc/

File: symphony-bdk-examples/bdk-app-spring-boot-example/build.gradle [hunk 3]
```diff
@@ -10,3 +10,9 @@ dependencies {
     implementation project(':symphony-bdk-spring:symphony-bdk-app-spring-boot-starter')
     implementation 'org.springframework.boot:spring-boot-starter-security'
 }
+
+uploadArchives.enabled = false
+
+bootJar {
```
Comment (symphony-youri) [inline symphony-bdk-examples/bdk-app-spring-boot-example/build.gradle:16]: what was the issue with bootJar?

**Comment by symphony-youri** [comment]
Isn't it a duplicate of what Snyk does?

File: symphony-bdk-examples/bdk-app-spring-boot-example/build.gradle (hunk 3 above)
Comment (symphony-thibault) [reply symphony-bdk-examples/bdk-app-spring-boot-example/build.gradle]: Nothing apparently...

File: build.gradle (hunk 2 above)
Comment (symphony-thibault) [reply build.gradle]: done

File: .circleci/config.yml (hunk 1 above)
Comment (symphony-thibault) [reply .circleci/config.yml]: yep, fixed

**Review by symphony-youri — APPROVED** [review]
I hope you'll squash the commits :) I don't want to leave my name everywhere

**Comment by symphony-thibault** [comment]
> @symphony-youri Isn't it a duplicate of what Snyk does?

I'd say that we have way more control over the licenses we want to allow or not with this Maven plugin. It could also be because we never configured Snyk to [reject some particular licenses](https://snyk.io/learn/open-source-licenses/).

**Comment by symphony-thibault** [comment]
@maoo you can find the Dependency License Report [here](https://1767-126734944-gh.circle-artifacts.com/0/~/reports/dependency-license/build/reports/dependency-license/index.html)

**Comment by maoo** [comment]
This is perfect @symphony-thibault , I'll use the list you shared above to run my checks. Thanks for the quick response!

**Pull request merged**